### ISSUE TITLE
Data race in TSI hll

### ISSUE THREAD

**Issue by jwilder** [influxdata/influxdb#7882]
Testing `tsi` branch `inmem` index I was able to trigger a data race panic:

```bash
$ go install -race  ./... && INFLUXDB_DATA_INDEX_VERSION=inmem influxd
$ influx-stress insert -n 100000000 --pps 1000000
```

```
WARNING: DATA RACE
Write at 0x00c4203e2180 by goroutine 320:
  runtime.mapassign1()
      /usr/local/go/src/runtime/hashmap.go:442 +0x0
  github.com/influxdata/influxdb/pkg/estimator/hll.(*Plus).Add()
      /Users/jason/go/src/github.com/influxdata/influxdb/pkg/estimator/hll/hll.go:114 +0x11a
  github.com/influxdata/influxdb/tsdb/index/inmem.(*Index).CreateSeriesIfNotExists()
      /Users/jason/go/src/github.com/influxdata/influxdb/tsdb/index/inmem/inmem.go:167 +0x6ea
  github.com/influxdata/influxdb/tsdb/index/inmem.(*ShardIndex).CreateSeriesIfNotExists()
      /Users/jason/go/src/github.com/influxdata/influxdb/tsdb/index/inmem/inmem.go:702 +0x103
  github.com/influxdata/influxdb/tsdb/index/inmem.(*ShardIndex).CreateSeriesListIfNotExists()
      /Users/jason/go/src/github.com/influxdata/influxdb/tsdb/index/inmem/inmem.go:694 +0x15a
  github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).CreateSeriesListIfNotExists()
      /Users/jason/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:884 +0xdb
  github.com/influxdata/influxdb/tsdb.(*Shard).validateSeriesAndFields()
      /Users/jason/go/src/github.com/influxdata/influxdb/tsdb/shard.go:583 +0x7f6
  github.com/influxdata/influxdb/tsdb.(*Shard).WritePoints()
      /Users/jason/go/src/github.com/influxdata/influxdb/tsdb/shard.go:439 +0x148
  github.com/influxdata/influxdb/tsdb.(*Store).WriteToShard()
      /Users/jason/go/src/github.com/influxdata/influxdb/tsdb/store.go:857 +0x184
  github.com/influxdata/influxdb/coordinator.(*PointsWriter).writeToShard()
      /Users/jason/go/src/github.com/influxdata/influxdb/coordinator/points_writer.go:350 +0x10c
  github.com/influxdata/influxdb/coordinator.(*PointsWriter).WritePoints.func1()
      /Users/jason/go/src/github.com/influxdata/influxdb/coordinator/points_writer.go:307 +0xa7

Previous read at 0x00c4203e2180 by goroutine 11:
  [failed to restore the stack]

Goroutine 320 (running) created at:
  github.com/influxdata/influxdb/coordinator.(*PointsWriter).WritePoints()
      /Users/jason/go/src/github.com/influxdata/influxdb/coordinator/points_writer.go:308 +0x42b
  github.com/influxdata/influxdb/services/httpd.(*Handler).serveWrite()
      /Users/jason/go/src/github.com/influxdata/influxdb/services/httpd/handler.go:660 +0x9d3
  github.com/influxdata/influxdb/services/httpd.(*Handler).(github.com/influxdata/influxdb/services/httpd.serveWrite)-fm()
      /Users/jason/go/src/github.com/influxdata/influxdb/services/httpd/handler.go:130 +0x69
  github.com/influxdata/influxdb/services/httpd.authenticate.func1()
      /Users/jason/go/src/github.com/influxdata/influxdb/services/httpd/handler.go:883 +0xbdc
  net/http.HandlerFunc.ServeHTTP()
      /usr/local/go/src/net/http/server.go:1726 +0x51
  github.com/influxdata/influxdb/services/httpd.(*Handler).responseWriter.func1()
      /Users/jason/go/src/github.com/influxdata/influxdb/services/httpd/handler.go:1087 +0xc2
  net/http.HandlerFunc.ServeHTTP()
      /usr/local/go/src/net/http/server.go:1726 +0x51
  github.com/influxdata/influxdb/services/httpd.gzipFilter.func1()
      /Users/jason/go/src/github.com/influxdata/influxdb/services/httpd/handler.go:1017 +0x22b
  net/http.HandlerFunc.ServeHTTP()
      /usr/local/go/src/net/http/server.go:1726 +0x51
  github.com/influxdata/influxdb/services/httpd.cors.func1()
      /Users/jason/go/src/github.com/influxdata/influxdb/services/httpd/handler.go:1061 +0x131
  net/http.HandlerFunc.ServeHTTP()
      /usr/local/go/src/net/http/server.go:1726 +0x51
  github.com/influxdata/influxdb/services/httpd.requestID.func1()
      /Users/jason/go/src/github.com/influxdata/influxdb/services/httpd/handler.go:1071 +0x1b3
  net/http.HandlerFunc.ServeHTTP()
      /usr/local/go/src/net/http/server.go:1726 +0x51
  github.com/influxdata/influxdb/services/httpd.(*Handler).logging.func1()
      /Users/jason/go/src/github.com/influxdata/influxdb/services/httpd/handler.go:1079 +0x13f
  net/http.HandlerFunc.ServeHTTP()
      /usr/local/go/src/net/http/server.go:1726 +0x51
  github.com/influxdata/influxdb/services/httpd.(*Handler).recovery.func1()
      /Users/jason/go/src/github.com/influxdata/influxdb/services/httpd/handler.go:1104 +0x198
  net/http.HandlerFunc.ServeHTTP()
      /usr/local/go/src/net/http/server.go:1726 +0x51
  github.com/bmizerany/pat.(*PatternServeMux).ServeHTTP()
      /Users/jason/go/src/github.com/bmizerany/pat/mux.go:117 +0x85d
  github.com/influxdata/influxdb/services/httpd.(*Handler).ServeHTTP()
      /Users/jason/go/src/github.com/influxdata/influxdb/services/httpd/handler.go:258 +0x334
  net/http.serverHandler.ServeHTTP()
      /usr/local/go/src/net/http/server.go:2202 +0xbb
  net/http.(*conn).serve()
      /usr/local/go/src/net/http/server.go:1579 +0x5f6

Goroutine 11 (running) created at:
  github.com/influxdata/influxdb/monitor.(*Monitor).Open()
      /Users/jason/go/src/github.com/influxdata/influxdb/monitor/service.go:125 +0x484
  github.com/influxdata/influxdb/cmd/influxd/run.(*Server).Open()
      /Users/jason/go/src/github.com/influxdata/influxdb/cmd/influxd/run/server.go:435 +0x1457
  github.com/influxdata/influxdb/cmd/influxd/run.(*Command).Run()
      /Users/jason/go/src/github.com/influxdata/influxdb/cmd/influxd/run/command.go:117 +0xeb8
  main.(*Main).Run()
      /Users/jason/go/src/github.com/influxdata/influxdb/cmd/influxd/main.go:89 +0x24d
  main.main()
      /Users/jason/go/src/github.com/influxdata/influxdb/cmd/influxd/main.go:46 +0xe0
```

**Comment by e-dard** [comment]
Part of this (for `inmem` indexes) is fixed via ab94c1b.

**Comment by e-dard** [comment]
Should be resolved.
